pkg/util/strings.go

```diff
@@ -139,3 +140,80 @@ func MergeSortedSlices(ctx context.Context, a ...[]string) ([]string, error) {
 	}
 	return r, nil
 }
+
+type Interner interface {
+	Intern(s string) string
+	Release(s string)
+}
+
+// NewInterner returns a new Interner to be used to intern strings.
+// Based on https://github.com/prometheus/prometheus/blob/726ed124e4468d0274ba89b0934a6cc8c975532d/storage/remote/intern.go#L51
+func NewInterner() Interner {
```

Comment:
I wonder if we should ask Prometheus to make this public instead?

Reply:
We could but as this is experimental for now I did not want to do. If we really see benefit on this we can propose Prometheus to expose it .